### PR TITLE
feat: Add compiled version info

### PR DESCRIPTION
**Pull request by paulgessinger**
This should help us detect when we're running with a mismatched header/shared lib combo, like can happen in the hack-build in Athena.

### CODE REVIEW

File: Core/ActsVersion.hpp.in [hunk 1]
```diff
@@ -26,4 +26,23 @@ constexpr unsigned int Version =
...
 constexpr const char* const CommitHashShort = "@_acts_commit_hash_short@";
 
+struct VersionInfo {
+  unsigned int major;
+  unsigned int minor;
+  unsigned int patch;
+  const char* const commitHash;
+  const char* const commitHashShort;
+
+  VersionInfo();
```
Comment (andiwand) [inline Core/ActsVersion.hpp.in:36]: can this be private?

File: Core/ActsVersion.hpp.in [hunk 2]
```diff
@@ -26,4 +26,23 @@ constexpr unsigned int Version =
...
+
+  static bool isConsistent() {
```
Comment (andiwand) [inline Core/ActsVersion.hpp.in:38]: can this be `constexpr` or be implemented in the source file?

Comment (paulgessinger) [reply Core/ActsVersion.hpp.in:38]: The idea is that it is implement in the header. This way I can check if the shared lib and the header that some compilation unit was looking at report the same version.

File: Core/ActsVersion.hpp.in (hunk 1 above)
Comment (paulgessinger) [reply Core/ActsVersion.hpp.in:36]: Probably, yeah.

**Review by andiwand — DISMISSED** [review]

File: Core/ActsVersion.hpp.in [hunk 3]
```diff
@@ -26,4 +28,26 @@ constexpr unsigned int Version =
 constexpr const char* const CommitHash = "@_acts_commit_hash@";
 constexpr const char* const CommitHashShort = "@_acts_commit_hash_short@";
 
+struct VersionInfo {
+  unsigned int versionMajor;
+  unsigned int versionMinor;
+  unsigned int versionPatch;
+  const char* const commitHash;
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/ActsVersion.hpp.in:35]: I don't think we do. I'll delete the default constructor to be sure.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/5951612)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291247) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291246) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291245) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291244) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291243) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291242) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291241) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291240) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291239) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291238) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291237) |
|:red_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291236) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291235) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291234) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291233) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291232) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31291231) |

**Pull request merged**